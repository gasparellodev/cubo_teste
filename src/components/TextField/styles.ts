import styled, { css, DefaultTheme } from 'styled-components'

type WrapperProps = {
  error: boolean
}

const wrapperModifiers = {
  error: (theme: DefaultTheme) => css`
    border: 0.1rem solid ${theme.colors.feedback.negativePure};
  `
}

export const Wrapper = styled.div<WrapperProps>`
  ${({ theme, error }) => css`
    display: flex;
    align-items: center;
    background: transparent;
    border: 0.1rem solid #555555;
    border-radius: ${theme.border.radius.medium};
    width: 100%;
    max-width: 44.5rem;
    position: relative;

    &:focus-within {
      background: ${theme.colors.lightgrey};
    }
    &::backdrop{
      background-color: red;
    }

    ${error && wrapperModifiers.error(theme)}
  `}
`

export const Input = styled.input<{ error: boolean }>`
  ${({ theme, error }) => css`
    //color: ${theme.colors.white};
    font-family: ${theme.font.family};
    font-size: ${theme.font.content.regular};
    font-weight: ${theme.font.weights.medium};
    padding: 2rem 3.3rem;
    background: transparent;
    border: 0;
    outline: none;
    width: 100%;

    &::placeholder {
      color: ${theme.colors.black};
    }

    &:-webkit-autofill {
      border-radius: ${theme.border.radius.medium};

      filter: none;

      &,
      &:active,
      &:focus,
      &:hover {
        -webkit-box-shadow: 0 0 0 5rem ${theme.colors.base.blueDark} inset !important;
      }

      -webkit-text-fill-color: white !important;
      font-family: ${theme.font.family};
      font-size: ${theme.font.content.regular};

      &::first-line {
        font-family: ${theme.font.family};
        font-size: ${theme.font.content.regular};
        color: #fff;
      }
    }

    ${error &&
    css`
      padding-right: 21rem;
    `}
  `}
`

export const WrapperError = styled.div`
  display: flex;
  align-items: center;
  position: absolute;
  right: 2rem;
  top: 50%;
  transform: translateY(-50%);
`

export const ErrorMessage = styled.span`
  ${({ theme }) => css`
    color: ${theme.colors.feedback.negativePure};
    font-size: ${theme.font.content.xsmall};
    padding-right: 0.8rem;
    white-space: nowrap;
  `}
`

export const Icon = styled.div`
  display: flex;
  order: 1;
`
