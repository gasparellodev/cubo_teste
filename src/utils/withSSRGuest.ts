import {
  GetServerSideProps,
  GetServerSidePropsContext,
  GetServerSidePropsResult
} from 'next'
import { parseCookies } from 'nookies'
import { COOKIE_TOKEN } from 'services/auth'

export function withSSRGuest<P>(
  fn: GetServerSideProps<P>,
  destination = '/diagnostico'
): GetServerSideProps {
  return async (
    ctx: GetServerSidePropsContext
  ): Promise<GetServerSidePropsResult<P>> => {
    const { [COOKIE_TOKEN]: token } = parseCookies(ctx)

    // if (token) {
    //   return {
    //     redirect: {
    //       destination,
    //       permanent: false
    //     }
    //   }
    // }

    const resultFn = await fn(ctx)
    return resultFn
  }
}
