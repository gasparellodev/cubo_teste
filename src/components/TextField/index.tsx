import {
  InputHTMLAttributes,
  ForwardRefRenderFunction,
  forwardRef
} from 'react'

import ErrorIcon from 'components/icons/ErrorIcon'

import * as S from './styles'

export type TextFieldProps = {
  error?: string | boolean
} & InputHTMLAttributes<HTMLInputElement>

const TextField: ForwardRefRenderFunction<HTMLInputElement, TextFieldProps> = (
  { name, error = null, ...rest },
  ref
) => {
  return (
    <S.Wrapper error={!!error}>
      <S.Input type="text" name={name} error={!!error} {...rest} ref={ref} />
      {!!error && (
        <S.WrapperError>
          <S.ErrorMessage>{error}</S.ErrorMessage>
          <S.Icon data-testid="error-icon">
            <ErrorIcon />
          </S.Icon>
        </S.WrapperError>
      )}
    </S.Wrapper>
  )
}

export default forwardRef(TextField)
