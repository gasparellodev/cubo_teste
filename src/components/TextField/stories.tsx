import { Story, Meta } from '@storybook/react/types-6-0'

import TextField, { TextFieldProps } from '.'

export default {
  title: 'Form/TextField',
  component: TextField,
  args: {
    name: 'email'
  }
} as Meta<TextFieldProps>

export const Default: Story<TextFieldProps> = (args) => <TextField {...args} />

Default.args = {
  placeholder: 'Placeholder'
}

export const withError: Story<TextFieldProps> = (args) => (
  <TextField {...args} />
)

withError.args = {
  error: true,
  value: 'email@error'
}
