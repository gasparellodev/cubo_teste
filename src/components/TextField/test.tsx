import userEvent from '@testing-library/user-event'
import { render, screen, waitFor } from 'utils/test-utils'

import TextField from '.'

describe('<Checkbox />', () => {
  it('should render with placeholder', () => {
    const { container } = render(<TextField placeholder="Test" />)

    const errorIcon = screen.queryByTestId('error-icon')

    expect(errorIcon).not.toBeInTheDocument()
    expect(screen.getByPlaceholderText(/test/i)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render without placeholder', () => {
    render(<TextField />)

    const input = screen.getByRole('textbox')

    expect(input).not.toHaveAttribute('placeholder')
  })

  it('should render an icon when error props is passed', () => {
    render(<TextField error />)

    const errorIcon = screen.getByTestId('error-icon')

    expect(errorIcon).toBeInTheDocument()
  })

  it('should change its value when typing', async () => {
    render(<TextField />)

    const input = screen.getByRole('textbox')
    const text = 'This is my new text'
    userEvent.type(input, text)

    await waitFor(() => {
      expect(input).toHaveValue(text)
    })
  })

  it('should be accessible with tab', () => {
    render(<TextField />)

    expect(document.body).toHaveFocus()

    userEvent.tab()

    expect(screen.getByRole('textbox')).toHaveFocus()
  })
})
