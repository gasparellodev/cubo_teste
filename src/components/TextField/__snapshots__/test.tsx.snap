// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> should render with placeholder 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: transparent;
  border: 0.1rem solid #555555;
  border-radius: 0.6rem;
  width: 100%;
  max-width: 44.5rem;
  position: relative;
}

.c0:focus-within {
  background: #253245;
}

.c1 {
  color: #FFFFFF;
  font-family: 'Poppins',-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-size: 1.6rem;
  font-weight: 500;
  padding: 2rem 3.3rem;
  background: transparent;
  border: 0;
  outline: none;
  width: 100%;
}

.c1::-webkit-input-placeholder {
  color: #FFFFFF;
}

.c1::-moz-placeholder {
  color: #FFFFFF;
}

.c1:-ms-input-placeholder {
  color: #FFFFFF;
}

.c1::placeholder {
  color: #FFFFFF;
}

.c1:-webkit-autofill {
  border-radius: 0.6rem;
  -webkit-filter: none;
  filter: none;
  -webkit-text-fill-color: white !important;
  font-family: 'Poppins',-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-size: 1.6rem;
}

.c1:-webkit-autofill,
.c1:-webkit-autofill:active,
.c1:-webkit-autofill:focus,
.c1:-webkit-autofill:hover {
  -webkit-box-shadow: 0 0 0 5rem #253245 inset !important;
}

.c1:-webkit-autofill::first-line {
  font-family: 'Poppins',-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-size: 1.6rem;
  color: #fff;
}

<div
  class="c0"
>
  <input
    class="c1"
    placeholder="Test"
    type="text"
  />
</div>
`
